async-nats/src/jetstream/consumer/pull.rs

```diff
@@ -246,7 +252,110 @@ impl<'a> futures::Stream for Sequence<'a> {
     }
 }
 
-pub type Stream<'a> = stream::TryFlatten<Sequence<'a>>;
+pub struct Stream<'a> {
+    pending_messages: usize,
+    subscriber: Subscriber,
+    context: Context,
+    inbox: String,
+    subject: String,
+    batch_config: BatchConfig,
```

Comment:
Will this mutate? otherwise we can serialize the batch config once on construction.

Reply:
Not by default, but the Wally approach would mutate the request.